src/algebra/bit manipulation.md

```diff
@@ -0,0 +1,214 @@
+---
+tags:
+  - Translated
+e_maxx_link: all_submask
+---
+
+# Bit operators
+
+When finding solutions to problems, a very important factor is the simplicity and the execution time. Now, in this bitwise operations are very important and you can take advantage of them.
+
+* ## Bit shift operators
+### Multiplication and division by powers of 2
+
+There are two operators for shifting bits.
+
+**>>** One bit to the right. Which would be the same as a division by a power of two.
+
+**<<** One bit to the left. And this would be a multiplication, equally by a power of two.
+
+Let's see an example.
+
+Let's say we have a number $N$, which. $N = 4$.
+
+The binary representation of **4** is:
+
+$4_2 = 100$
+
+If we shift one bit to the left, then we would be adding a bit to the end, in this case a 0.
+
+$4_2 << 1 = 100 << 1 = 1000$
```

Comment:
Ok, it's really not important, the idea is clear. Although I think it might confuse the symbology a bit, I don't think you find many sources that show those operators that way. But annoying I would like to show it as it is, I think it is clearer for the reader. But annoying I would like to show it as it is, I think it is clearer for the reader. But if not It does not matter, it is changed and that's it.

Reply:
Ok, but I would like to show it that way to avoid confuciations with the symbology. I don't think you find many sources that show it that way. Personally I get a little lost, but anyway, it doesn't matter if you prefer it that way, it changes.If you prefer it to be that way, it is changed